docs/library/machine.QECNT.rst

```diff
@@ -0,0 +1,379 @@
+   If the total position range is still too small, you can create your own cycles
+   counter using the irq() callback method.
+
+.. method:: Encoder.irq(trigger=event, value=nnn, handler=handler, hard=False)
+
+   Specifies, that the *handler* is called when the respective *event* happens.
+
+   *event* may be:
+    - Encoder.IRQ_MATCH Triggered when the position counter matches the match value.
+    - Encoder.IRQ_ROLL_OVER Triggered when the position counter rolls over from the highest
+      to the lowest value.
+    - Encoder.IRQ_ROLL_UNDER Triggered when the position counter rolls under from the lowest
+      to the highest value.
+
+   The callback is called, when the Encoder is at *value*. For fas signals, the actual counter 
+   value may be different from the trigger value. 
+   The callback function *handler* receives a single argument, which is the Encoder object. All
+   events share the same callback. The event which triggers the callback can be identified
+   with the Encoder.status() method. The argument *hard* specifies, whether the callback is called
+   as a hard interrupt or as regular scheduled function. Hard interrupts have always a short latency,
+   but are limited in that they must not allocate memory. Regular scheduled functions are not limited
+   in what can be used, but depending on the load of the device execution may be delayed.
+   Under low load, the difference in latency is minor.
+
+   The default arguments values are trigger=0, handler=None, hard=False. The callback will be
+   disabled, when called with handler=None.
+
+   The position match event is triggered as long as the position and match value are identical.
+   Therefore the position match callback is run in a one-shot fashion, and has to be enabled
+   again when the position has changed.
+
+.. method:: Encoder.status()
+
+   Returns the event status flags of the recent handled Encoder interrupt as a bitmap.
+   The assignment of events to the bits are:
+
+   - 0: Transition at the HOME signal. (*)
+   - 1: Transition at the INDEX signal. (*)
+   - 2: Watchdog event. (*)
+   - 3 or Encoder.IRQ_MATCH: Position match event.
+   - 4: Phase_A and Phase_B changed at the same time. (*)
+   - 5 or Encoder.IRQ_ROLL_OVER: Roll-Over event of the counter.
+   - 6 or Encoder.IRQ_ROLL_UNDER: Roll-Under event of the counter.
+   - 7: Direction of the last count. 1 for counting up, 0 for counting down.
+
+   (*) These flags are defined, but not (yet) enabled.
+
+.. method:: Encoder.id()
+
+   Return the id of the Encoder. That may be helpful for interrupt callback functions.
+
+The Encoder was tested to work up to 25MHz on a Teensy. It may work at
+higher frequencies as well, but that was the limit of the test set-up.
+
+
+.. _mimxrt_machine.Counter:
+
+class Counter-- Signal counter for i.MXRT MCUs
+==============================================
+
+This class provides a Counter service using the Quadrature Encoder module
```

Comment:
I think I have seen a couple of forum threads on the development approaches on these Encoder/Counter Classes, but as a new Micropython users, looking at the docs it is not clear why a second class exists and when to use one of these services over another.

Reply:
Encoder and Counter behave different. The Encoder requires at least in this implementation the two-phase signal, the counter works on a single phase.